loader/vk_loader_platform.h

```diff
@@ -100,6 +99,26 @@ static inline bool loader_platform_is_path_absolute(const char *path) {
 
 static inline char *loader_platform_dirname(char *path) { return dirname(path); }
 
+#if defined(__linux__)
+
+// find application path + name. Path cannot be longer than 1024, returns NULL if it is greater than that.
+static inline char *loader_platform_executable_path(char buffer[1024]) {
```

Comment:
Why is the buffer restricted to size 1024? This seems like a silly constraint to impose. It makes far more sense to just take an extra argument that gives the size (like every platform implementation being called here does).